src/snowflake/snowpark/_internal/ast.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import base64
+import itertools
+import sys
+import uuid
+from dataclasses import dataclass
+from typing import Callable, Tuple
+
+import snowflake.snowpark._internal.proto.generated.ast_pb2 as proto
+from snowflake.snowpark.version import VERSION
+
+# The current AST version number (generated by the DSL).
+CLIENT_AST_VERSION = proto.__Version__.MAX_VERSION
+
+
+@dataclass
+class TrackedCallable:
```

Comment:
can you add comment to explain what this is used for?

Reply:
Added docstring.